java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java

```diff
@@ -6605,6 +6608,30 @@ public int setPrimaryInterface(User loggedInUser, Integer serverId,
         return 1;
     }
 
+    /**
+     * Sets new primary FQDN
+     * @param loggedInUser The current user
+     * @param serverId Server ID
+     * @param fqdn Primary FQDN
+     * @return 1 if success, exception thrown otherwise
+     * @throws Exception If FQDN does not exist Exception is thrown
+     *
+     * @xmlrpc.doc Sets new primary FQDN
+     * @xmlrpc.param #param("string", "sessionKey")
+     * @xmlrpc.param #param("int", "serverId")
+     * @xmlrpc.param #param("string", "fqdn")
+     * @xmlrpc.returntype #return_int_success()
+     */
+    public int setPrimaryFqdn(User loggedInUser, Integer serverId, String fqdn) {
+        Server server = lookupServer(loggedInUser, serverId);
+
+        if (!server.existsFqdn(fqdn)) {
```

Comment:
Would it be more straightforward to have a finder method that returns an `Optional<ServerFQDN>` instance and set the flag there if it is found?